dp/CoinChange.py

```diff
@@ -0,0 +1,31 @@
+//Given a value N, if we want to make change for N Cents, and we have infinite supply of each of S(coins) = { S1, S2, .. , Sm}, 
+//how many ways can we make the change? 
+
+//For example, for N = 4 and S = {1,2,3}, there are four solutions: {1,1,1,1},{1,1,2},{2,2},{1,3}. So output should be 4.
+//For N = 10 and S = {2, 5, 3, 6}, there are five solutions: {2,2,2,2,2}, {2,2,3,3}, {2,2,6}, {2,3,5} and {5,5}
+
+coins = map(int, raw_input().split(','))
+n = int(input())
```

Comment:
please use `__name__=="__main__"` or wrap this in a function so it prevents global variables and it is easier to see the order of execution.